src/UI/Cells/SelectAllCell.js

```diff
@@ -38,7 +40,11 @@ module.exports = BackgridSelectAll.extend({
             model.trigger('backgrid:select', model, checked);
         });
 
-        this.model.collection.lastToggled = undefined;
-        this.model.collection.checked = undefined;
+        collection.lastToggled = undefined;
+        collection.checked = undefined;
+    },
+
+    _getCollection : function() {
+        return this.column.get('sortedCollection') || this.model.collection;
```

Comment:
Inline this method and pass the collection to _selectRange instead of getting it again.